runtime/src/main/java/org/wildfly/swarm/config/runtime/invocation/ListTypeAdapter.java

```
@@ -16,65 +16,28 @@
-                        addDmrValueTo(modelMode.get(detypedName), listValueType, v);
-                    }
-            );
-        }
-
-    }
-
-    private void addDmrValueTo(ModelNode target, ModelType type, Object propValue)
-    {
-        if(type.equals(ModelType.STRING))
-        {
-            target.add((String) propValue);
-        }
-        else if(type.equals(ModelType.INT))
-        {
-            target.add((Integer) propValue);
-        }
-        else if(type.equals(ModelType.DOUBLE))
-        {
-            target.add((Double) propValue);
-        }
-        else if(type.equals(ModelType.LONG))
-        {
-            target.add((Long) propValue);
-        }
-        else if(type.equals(ModelType.BOOLEAN))
-        {
-            target.add((Boolean) propValue);
-        }
-        else if (type.equals(ModelType.OBJECT)) {
-            ModelNode val = target.addEmptyObject();
-            Map<String,?> map = (Map) propValue;
-
-            for (String key : map.keySet()) {
-                Object value = map.get(key);
-                if ( value instanceof String ) {
-                    val.get( key ).set( (String) value );
-                } else if ( value instanceof Boolean ) {
-                    val.get( key ).set( (Boolean) value );
+            for (Object value : list) {
+                ModelNode node = target.add();
+                if (value instanceof List) {
+                    new ListTypeAdapter().toDmr(node, (List) value);
+                } else if (value instanceof Map) {
+                    new MapTypeAdapter().toDmr(node, (Map) value);
+                } else {
+                    ModelType type = Types.resolveModelType(value.getClass());
+                    new SimpleTypeAdapter().toDmr(node, type, value);
                 }
             }
         }
-        else
-        {
-            throw new RuntimeException("Unsupported DMR type: "+type);
-        }
-    }
 
+    }
 
+    // TODO handle composite values
```

I admit I was just lazy here. Is the `fromDmr` method even used somewhere?

---

not really. I did that initially . The one thing you could do is to throw an exception indicating that it's not implemented, rather than silently ignoring this case